src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,179 @@
+package duke.storage;
+
+import duke.dukeexception.DukeException;
+import duke.task.TaskList;
+import duke.task.Todo;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Repeat;
+import duke.task.DoAfter;
+import duke.task.FixedDuration;
+import duke.ui.Ui;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Represents a storage to store the task list into a text file.
+ */
+public class Storage {
+    //protected String filePath = "./";
+    protected String filePath = "";   //27-28, 40-47
+    String storageClassPath = Storage.class.getProtectionDomain().getCodeSource().getLocation().getPath();
+    private static final int ZERO = 0;
```

Comment:
If these constants are repeatedly used by many classes, you can define it as an enum to increase the readability and maintainability